cloudbuild.yaml

```diff
@@ -23,6 +23,15 @@ steps:
     gsutil ls gs://$PROJECT_ID-tfstate || gsutil mb gs://$PROJECT_ID-tfstate
     (gsutil versioning get gs://$PROJECT_ID-tfstate | grep Enabled) || gsutil versioning set on gs://$PROJECT_ID-tfstate
 
+- id: 'get machine type'
+  name: 'gcr.io/cloud-builders/gcloud'
+  entrypoint: 'bash'
+  args:
+    - '-c'
+    - |-
+      apt -qqy update && apt -qqy install jq
+      (gcloud --format=json compute project-info describe | jq -e -r '.commonInstanceMetadata.items[] | select(.key == "radlab-machine-type") | .value' || printf "n2-standard-32") > /workspace/radlab_machine_type
+
```

Comment:
Did you consider curl'ing http://metadata.google.internal/computeMetadata/v1 instead? It could provide a friendlier integration with the CLI.

Reply:
Actually that was my initial idea which unfortunately didn't work. This piece of information is kind of hard to find but it seems that Google Cloud Build workers cannot talk to the "real" metadata server; instead they [talk to a spoofed one](https://github.com/GoogleCloudPlatform/cloud-builders/issues/80#issuecomment-518933112).

Reply:
Alternatively maybe we could map this to `substitutions` in the clould build trigger installed during the bootstrap phase:
https://github.com/proppy/rad-lab-deploy/blob/tuning/bootstrap/main.tf#L88

Or find a way to set some project level state that we can then `tf import` (just like we do for the project id):
https://github.com/proppy/rad-lab-deploy/blob/tuning/cloudbuild.yaml#L47

Or should we generalize being import to infer all terraform variables from the `commonInstanceMetadata`?

Reply:
Mapping this as a substitution in the trigger would be the elegant solution, however considering that the bootstrapping scripts are usually executed only once I think that this approach could prove to be a little bit inflexible.

Regarding the project-level state and accessing that from Terraform -- I attempted to do something of this sort before resorting to hacking around with `gcloud` and `jq`, however I could not find any suitable data source which would allow me to access e.g. project metadata. The closest that I could find was the [google_tags_tag_value](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/tags_tag_value) data source but this works only on the organization level (and we'd need project level).

There's also another way that I've just found: instead of using `gcloud` we could add the [google_compute_project_metadata_item](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_project_metadata_item) resource, though personally I think that it is a little bit of an overkill.

I think that generalizing it is a good idea! This could work in such a way that we'd extract all variables starting with `radlab-` and then process them to extract variables that match the names with [TF variables](https://github.com/proppy/rad-lab-deploy/blob/tuning/variables.tf). Keep in mind that given that we only have 2 variables for now (+ the one that this PR adds), I would suggest implementing it once we have more parameters.

Reply:
> There's also another way that I've just found: instead of using gcloud we could add the google_compute_project_metadata_item resource, though personally I think that it is a little bit of an overkill.

I think that could work, and allow us to not have the default be defined in two places:
- cloud build fallback
- module input var def https://github.com/proppy/rad-lab-deploy/blob/tuning/rad-lab/modules/silicon_design/variables.tf#L158